dotCMS/src/main/java/com/dotcms/rest/api/v1/template/TemplateResource.java

```diff
@@ -0,0 +1,752 @@
+     *
+     * where:
+     *
+     * <ul>
+     * <li>filter-string: just return Template who content this pattern into its title</li>
+     * <li>page: page to return</li>
+     * <li>per_page: limit of items to return</li>
+     * <li>ordeby: field to order by</li>
+     * <li>direction: asc for upward order and desc for downward order</li>
+     * <li>host: filter by host's id</li>
+     * </ul>
+     *
+     * Url example: v1/templates?filter=test&page=2&orderby=title
+     *
+     * @param httpRequest
+     * @return
+     */
+    @GET
+    @JSONP
+    @NoCache
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response list(@Context final HttpServletRequest httpRequest,
+                                        @Context final HttpServletResponse httpResponse,
+                                        @QueryParam(PaginationUtil.FILTER)   final String filter,
+                                        @QueryParam(PaginationUtil.PAGE)     final int page,
+                                        @QueryParam(PaginationUtil.PER_PAGE) final int perPage,
+                                        @DefaultValue("title") @QueryParam(PaginationUtil.ORDER_BY) final String orderBy,
+                                        @DefaultValue("ASC") @QueryParam(PaginationUtil.DIRECTION)  final String direction,
+                                        @QueryParam(ContainerPaginator.HOST_PARAMETER_ID)           final String hostId) {
+
+        final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                .requestAndResponse(httpRequest, httpResponse).rejectWhenNoUser(true).init();
+        final User user = initData.getUser();
+        final Optional<String> checkedHostId = Optional.ofNullable(Try.of(()-> APILocator.getHostAPI()
+                .find(hostId, user, false).getIdentifier()).getOrNull());
+
+        Logger.debug(this, ()-> "Getting the List of templates");
+
+        final Map<String, Object> extraParams = Maps.newHashMap();
+        checkedHostId.ifPresent(checkedHostIdentifier -> extraParams.put(ContainerPaginator.HOST_PARAMETER_ID, checkedHostIdentifier));
+        return this.paginationUtil.getPage(httpRequest, user, filter, page, perPage, orderBy, OrderDirection.valueOf(direction),
+                extraParams);
+    }
+
+    /**
+     * Return a {@link com.dotmarketing.portlets.templates.model.Template} based on the inode
+     *
+     * @return Response
+     */
+    @GET
+    @Path("/{templateInode}")
+    @JSONP
+    @NoCache
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response getByInode(@Context final HttpServletRequest  httpRequest,
+                               @Context final HttpServletResponse httpResponse,
+                               @PathParam("templateInode") final String templateInode) throws DotSecurityException, DotDataException {
```

Comment:
Codacy found an issue: [The String literal "templateInode" appears 6 times in this file; the first occurrence is on line 216](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6042990)